app/src/main/java/ch/epfl/favo/notifications/FirebaseMessagingService.java

```diff
@@ -43,34 +51,42 @@ public static void showNotification(
             .setAutoCancel(true)
             .setSound(defaultSoundUri)
             .setContentIntent(pendingIntent);
+
     NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
+
     // Since android Oreo notification channel is needed
     if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
       NotificationChannel channel =
           new NotificationChannel(channelId, CHANNEL_NAME, NotificationManager.IMPORTANCE_DEFAULT);
       Objects.requireNonNull(notificationManager).createNotificationChannel(channel);
     }
+
     Objects.requireNonNull(notificationManager)
         .notify(new Random().nextInt(), notificationBuilder.build());
   }
+
   // method called when new message (notification or data message) is received
   @Override
   public void onMessageReceived(RemoteMessage remoteMessage) {
-    // Check if message contains a data payload.
-    // if (remoteMessage.getData().size() > 0) {
-    // Log.d("FirebaseMessagingService", "Message data payload: " + remoteMessage.getData());
-    // do something with message data, like chat (TODO for later sprints)
-    // }
     // Check if message contains a notification payload.
     if (remoteMessage.getNotification() != null) {
       showNotification(this, remoteMessage, getString(R.string.default_notification_channel_id));
     }
   }
+
   // onNewToken callback fires whenever a new token is generated
+  @RequiresApi(api = Build.VERSION_CODES.N)
   @Override
   public void onNewToken(@NonNull String token) {
-    // Log.d(TAG, "Refreshed token: " + token);
-    // TODO send new refreshed token to db
-    // simply set the new notificationID for the current user and sync with the db
+    // update user notification id
+    UserUtil.getSingleInstance()
```

Comment:
Seems like this method is very similar to the one in the LoginActivity. Maybe we can extract some of it in UserUtil

Reply:
I'm not sure if there's a good way to factor a method out of the two: I would keep them separate because the one in SignInActivity posts a new user along with all the new/updated information while this one just updates a single field. It's possible to extract some part of this but maybe the utility would be minimal.